src/Aurora.Shared/Models/ValueOrNull.cs

```diff
@@ -34,6 +34,20 @@ public TResult Resolve<TResult>(Func<T, TResult> onValue, Func<string, TResult>
         return result;
     }
 
+    public Task ResolveAsync(Func<T, Task> onValue, Func<string, Task> onNull)
+    {
+        Task result;
+        if (HasValue)
+        {
+            result = onValue(Value!);
+        }
+        else
+        {
+            result = onNull.Invoke(NullMessage ?? "");
+        }
+        return result;
```

Comment:
and here could be returned just.Then and it looks more clear on my magazines

Reply:
introduced